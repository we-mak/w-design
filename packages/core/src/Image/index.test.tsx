import * as React from "react";
import { render, cleanup } from "@testing-library/react";
import { Provider, Image } from "..";

afterEach(cleanup);

describe("<Image/>", () => {
  const props = {
    "data-src": "https://source.unsplash.com/random",
    alt: "Test",
    width: "300",
    height: "400",
  };

  it("should render correct image", () => {
    const { container } = render(
      <Provider>
        <Image {...props} />
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });

  // describe('when the Image is out of view', ()=> {
  //   beforeAll(() => {
  //     const mockEntry = { isIntersecting: false }

  //   })
  // })
});
