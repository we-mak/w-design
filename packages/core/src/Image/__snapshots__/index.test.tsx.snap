// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Image/> should render correct image 1`] = `
<div>
  <img
    alt="Test"
    class="sc-AykKG jOTVdb"
    data-src="https://api.adorable.io/avatars/240/thunder"
    src="data:image/gif;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1HAwCAAAAC0lEQVR42mP88R8AAvUB+VkkrXoAAAAASUVORK5CYII="
  />
</div>
`;
